Rangetest.c

```diff
@@ -0,0 +1,21 @@
+#include <stdio.h>
+#include <assert.h>
+#include "Rangedetect.h"
+
+void TEST_CASE_01()
+{
+    int Data[]={4,5};
+    int Totalreadings=sizeof(Data)/sizeof(Data[0]);
+    assert(DetectAndDisplayRange(Data,Totalreadings) == 1);
+}
+void TEST_CASE_02()
+{
+    int Data[]={4,5,5,6,8,10};
+    int Totalreadings=sizeof(Data)/sizeof(Data[0]);
+    assert(DetectAndDisplayRange(Data,Totalreadings) == 3);
```

Comment:
Add assert for the content of the detected ranges as well. Try writing the test first, instead of referring to the code and then writing the test.